### ISSUE TITLE
[DOC] [RNG] Add documentation for RNG

### ISSUE THREAD

**Issue by mmeterel** [oneapi-src/oneMKL#201]
# Summary
RNG domain does not have documentation and needs to be added.

**Comment by zjin-lcf** [comment]
Could you please clarify if SYCL buffer examples are supported with RNG ?  Currently, the examples are written with USM. Thanks.

**Comment by zjin-lcf** [comment]
Could you please clarify if curand and cublas can be enabled together or only one of them can be enabled in the build instruction ?

**Comment by zjin-lcf** [comment]
Could you please clarify if ROCRAND is supported by HIPSYCL ?  If not, please add a note to the ROCRAND support in the build instruction.

```
  -DENABLE_ROCBLAS_BACKEND=True                     \
         -DTARGET_DOMAINS=blas                               # hipSYCL only supports the BLAS domain
```

**Comment by aelizaro** [comment]
Hi @zjin-lcf, thank you for noticing the issue! RocRand is supported with HIPSYCL you can use it right now with a build configuration similar to rocBLAS:
```
# Inside <path to onemkl>
mkdir build && cd build
cmake .. -DENABLE_CURAND_BACKEND=False                     \
         -DENABLE_MKLCPU_BACKEND=False/True              # hipSYCL supports MKLCPU backend
         -DENABLE_MKLGPU_BACKEND=False                       # disable Intel MKL GPU backend
         -DENABLE_ROCRAND_BACKEND=True                     \
         -DTARGET_DOMAINS=rng
         -DHIPSYCL_TARGETS=omp\;hip:gfx906                   # Specify the targeted device architectures
         -DONEMKL_SYCL_IMPLEMENTATION=hipSYCL                # Use the hipSYCL cmake integration
cmake --build .
ctest
cmake --install . --prefix <path_to_install_dir>
```

I prepared a PR with documentation fix #220 . I would be glad to help if you have any more questions.
